### DISCUSSION TITLE
How to create a team containing all members of an org?

### DISCUSSION THREAD

**Discussion by TyeMcQueen** [community/community#23469] [New to GitHub]
<p>The best practices for organizations help page suggests “we recommend creating an <code>all-employees</code> team and granting it appropriate access to all but the most top-secret repos”.  But that suggestion seems to ignore the significant problem of how to get every new org member added to such a team.</p>
<p>Note that I am not trying to exclude “org members” that are contractors so there is no need or desire to currate which org members get added to this team.  I am just trying to exclude some repos from org-wide read access.  gitlab supports not just public and private repos but “internal” repos that provide org-wide access (without having to apply this to all repos).</p>
<p>Having a pre-built “team” that is “entire org” would be helpful.</p>
<p>Thanks,</p>
<p>Tye</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/tyemcqueen">@tyemcqueen</a>,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>
<p>Cheers!</p>

**Comment by xitongsys** [comment]
<p>any update about this feature ?</p>

**Comment by ap0phi5** [comment]
<p>Also keen to see if this.</p>
<p>Though we could create an <strong>all-members</strong> team and add folks in bulk, future members would not automatically be added. I guess we could define a GH App to do this, though if there was a native capability, that would be preferential.</p>

**Comment by RobPasMue** [comment]
<p>At the Ansys org we have been recently working with this specific problem. We created an “all-members” team within our organization which intended to host all its members. We created a python script which runs every night through GitHub Actions (and using PyGitHub). This script basically detects if there have been any new member additions and adds them to the group. Feel free to take a look at it. It may solve your problem. :slight_smile:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/ansys/ansys-all-members">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/ansys/ansys-all-members" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/37798125/181077248-b4a0e0a7-14eb-4651-8239-c1435688fb27.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/4/0/40457f197869a62f06fa106a0f6b8dc26a1ec9aa_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/4/0/40457f197869a62f06fa106a0f6b8dc26a1ec9aa_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/4/0/40457f197869a62f06fa106a0f6b8dc26a1ec9aa.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/4/0/40457f197869a62f06fa106a0f6b8dc26a1ec9aa_2_10x10.png"></div>

<h3><a href="https://github.com/ansys/ansys-all-members" target="_blank" rel="noopener nofollow ugc">GitHub - ansys/ansys-all-members: Repository for updating the...</a></h3>

  <p>Repository for updating the ansys/all-members team automatically - GitHub - ansys/ansys-all-members: Repository for updating the ansys/all-members team automatically</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by philippmasino** [comment]
I would also like to grant read-access to all organization members, current or future, for some but not automatically for all repositories. the closest solutions are:
 i) I would have to create a team for that and then manage the team as suggested above. 
 ii) Or I could grant read access to ALL repositories, but i dont want that either, not everybody is allowed to read all repos. 
So there is a clear gap in the coverage of settings. For me, it would be intuitive if the organization itself could be added to the collaborators that have access via the teams or user dropdown menu (under github.com/organization/repo/settings/access)
